src/components/accounts/recovery_setup/SetupRecoveryMethod.js

```diff
@@ -40,27 +41,45 @@ class SetupRecoveryMethod extends Component {
         success: false,
         emailInvalid: false,
         phoneInvalid: false,
-        activeMethods: []
+        activeMethods: [],
+        hasFetchedMethods: false
     }
 
     componentDidMount() {
-        const { loadRecoveryMethods, accountId, router, recoveryMethods } = this.props;
+        const { router, getAccessKeys } = this.props;
         const { method } = router.location;
 
+        getAccessKeys()
+
         if (method) {
             this.setState({ option: method });
         }
 
-        if (recoveryMethods[accountId]) {
-            const confirmed = recoveryMethods[accountId].filter(method => method.confirmed)
+        this.setRecoveryMethods()
+    }
+
+    setRecoveryMethods = () => {
+        if (this.props.recoveryMethods[this.props.accountId]) {
+            const confirmed = this.props.recoveryMethods[this.props.accountId].filter(method => method.confirmed)
             this.setState({ activeMethods: confirmed.map(method => method.kind) });
         } else {
-            loadRecoveryMethods(accountId)
-                .then((payload) => {
-                    const confirmed = payload.data.filter(method => method.confirmed);
-                    this.setState({ activeMethods: confirmed.map(method => method.kind) });
-                })
+            if (!this.state.hasFetchedMethods) {
+                this.getMethods()
+            }
+        }
+    }
+
+    getMethods = async () => {
+
+        try {
+            await this.props.loadRecoveryMethods()
+        } catch(e) {
+            console.log(e)
+        } finally {
+            this.setState({ hasFetchedMethods: true })
```

Comment:
why `hasFetchedMethods` is set to `true` on error?

Reply:
@Patrick1904 some of your voodoo?

Reply:
Because we want to run this once only on mount. Do you have any thoughts on what we should do?

Reply:
@Patrick1904 error should set some specific error state?